tools/lsp/standalone.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+use crate::OutgoingRequestQueue;
+use core::cell::RefCell;
+use core::future::Future;
+use core::pin::Pin;
+use core::task::Poll;
+use lsp_types::notification::Notification;
+use std::rc::Rc;
+use std::sync::Arc;
+
+pub fn open(
+    args: super::Cli,
+    path: std::path::PathBuf,
+    component: Option<String>,
+) -> crate::common::Result<()> {
+    let cli_args = args.clone();
+    let lsp_thread = std::thread::Builder::new()
+        .name("LanguageServer".into())
+        .spawn(move || {
+            /// Make sure we quit the event loop even if we panic
+            struct QuitEventLoop;
+            impl Drop for QuitEventLoop {
+                fn drop(&mut self) {
+                    super::preview::quit_ui_event_loop();
+                }
+            }
+            let quit_ui_loop = QuitEventLoop;
+            if let Err(e) = fake_lsp(args, path, component) {
+                eprintln!("{e}");
+                std::process::exit(1);
+            }
+            drop(quit_ui_loop);
+        })
+        .unwrap();
+
+    super::preview::start_ui_event_loop(cli_args);
+    lsp_thread.join().unwrap();
+    Ok(())
+}
+
+fn fake_lsp(
+    args: super::Cli,
+    path: std::path::PathBuf,
+    component: Option<String>,
+) -> crate::common::Result<()> {
```

Comment:
Why do we need a fake language server?

It feels easier to extend the live-preview to request the files it did not find than to do an entire extra parsing run to find those... And that is the only functionality we really need from the LS at this point in time.